### PR TITLE
Warn users they cannot search by email address [OSF-5639]

### PR DESCRIPTION
**Pull request by mluke93**
## Purpose

When a user attempt to add contributors to a project, if the name entered is an email, we don't want to allow them to add it as an unregistered user (we want them to add "John Doe" not "jdoe89@gmail.com"). So we want to alert them to search by name, not email, when they search for an email address.

## Changes

The entered name is now checked against a Regular Expression to see if it fits the format of an email address (ie. [text]@[text].[text]). It then outputs the correct responses based on whether or not the name is an email or not.

## Side effects

None

## Ticket

https://openscience.atlassian.net/browse/OSF-5639

### CODE REVIEW

**Comment by mluke93** [comment]
Sara and I did a demo, and I changed the "error text" to her suggestion.

**Comment by brianjgeiger** [comment]
It would have been better to base off of the previous PR and then modified it with the feedback on that PR. At least pull in the additional tests from https://github.com/CenterForOpenScience/osf.io/pull/5695/files and ensure you're covering those cases as well.

File: website/templates/prereg_landing_page.mako [hunk 1]
```diff
@@ -61,7 +61,7 @@
       </div>
       %if has_draft_registrations:
       <div class="row">
-        <div class="prereg-button m-b-md p-md osf-box-lt p-md box-round" data-qtoggle-group="prereg" data-qtoggle-target="#existingPreregXS">Continue working on an existing preregistration</div>
+        <div class="prereg-button m-b-md p-md osf-box-lt p-md box-round" data-qtoggle-group="prereg" data-qtoggle-target="#existingPreregXS">Continue working on an existing draft preregistration</div>
```
Comment (brianjgeiger) [inline website/templates/prereg_landing_page.mako]: This doesn't look like it belongs to this PR.

File: website/templates/prereg_landing_page.mako [hunk 2]
```diff
@@ -98,7 +98,7 @@
             </td>
             %if has_draft_registrations:
             <td class="col-sm-${ num_cols } prereg-button-col">
-              <div class="prereg-button m-b-md p-md osf-box-lt p-md box-round" data-qtoggle-group="prereg" data-qtoggle-target="#existingPrereg">Continue working on an existing preregistration</div>
+              <div class="prereg-button m-b-md p-md osf-box-lt p-md box-round" data-qtoggle-group="prereg" data-qtoggle-target="#existingPrereg">Continue working on an existing draft preregistration</div>
```
Comment (brianjgeiger) [inline website/templates/prereg_landing_page.mako]: This doesn't look like it belongs to this PR.

**Comment by brianjgeiger** [comment]
Pass done. 🐧

**Comment by mluke93** [comment]
Sorry about the new PR. Wasn't sure how to handle it, so I went with the way I knew how. :P As far as the tests go, I copied everything that existed before I began. Those tests are included in with my additional tests. As far as the other things go... Those were from a different PR (https://github.com/CenterForOpenScience/osf.io/pull/5960). I'm not sure why they are here. Is it possible to remove/ignore them from github or do I need to go back to my local branch and remove those and recommit?

**Comment by brianjgeiger** [comment]
How about making a new feature branch from develop, then connect to Zach's remote and pull from his branch. Make whatever changes you need from there (which should just be the addition of a period if I remember correctly) and make sure it still works properly. Then make a PR from that.

**Comment by mluke93** [comment]
Alright, I'll go ahead and do that.

**Comment by brianjgeiger** [comment]
Still the old PR/branch.

**Pull request closed without merging**